### PR TITLE
Fix broken test on linux

### PR DESCRIPTION
**Pull request by mcollina**
Fixes https://github.com/nodejs/undici/issues/2697

### CODE REVIEW

**Review by KhafraDev — APPROVED** [review]

File: lib/fetch/index.js [hunk 1]
```diff
@@ -2021,7 +2021,9 @@ async function httpNetworkFetch (
       // into stream.
       const buffer = new Uint8Array(bytes)
       if (buffer.byteLength) {
-        fetchParams.controller.controller.enqueue(buffer)
+        try {
+          fetchParams.controller.controller.enqueue(buffer)
+        } catch {}
```
Comment (mcollina) [inline lib/fetch/index.js:2026]: @KhafraDev please review this. A wpt is failing without it: https://github.com/nodejs/undici/blob/main/test/wpt/tests/fetch/api/basic/stream-safe-creation.any.js

```
Uncaught exception in "D:\a\undici\undici\test\wpt\tests\fetch\api\basic\stream-safe-creation.any.js":
TypeError [ERR_INVALID_STATE]: Invalid state: ReadableStream is already closed
    at ReadableByteStreamController.enqueue (node:internal/webstreams/readablestream:1160:13)
    at fetchParams.controller.resume (D:\a\undici\undici\lib\fetch\index.js:2024:43)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
================================================================================================
Uncaught exception in "D:\a\undici\undici\test\wpt\tests\fetch\api\basic\stream-safe-creation.any.js":
TypeError [ERR_INVALID_STATE]: Invalid state: ReadableStream is already closed
    at ReadableByteStreamController.enqueue (node:internal/webstreams/readablestream:1160:13)
    at fetchParams.controller.resume (D:\a\undici\undici\lib\fetch\index.js:2024:43)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
================================================================================================
```

**Review by ronag — APPROVED** [review]

**Comment by mcollina** [comment]
@KhafraDev please take a look at the changes in fetch(), I'm not 100% convinced but the WPTs and our test pass.

**Review by KhafraDev — COMMENTED** [review]
I have 0 time this week to dig into it, but nothing seems off to me.

**Comment by Uzlopak** [comment]
I will have a look now

**Pull request merged**